x-pack/plugins/enterprise_search/server/lib/route_config_helpers.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  DestructiveRouteMethod,
+  RouteConfig,
+  RouteConfigOptions,
+  RouteMethod,
+  RouteValidatorFullConfig,
+} from 'kibana/server';
+
+import { schema } from '@kbn/config-schema';
+
+type Config<Method extends DestructiveRouteMethod> = RouteConfig<unknown, unknown, unknown, Method>;
+
+// We disallow options to set 'body' because we override them.
+interface ConfigWithoutBodyOptions<P, Q, B, Method extends RouteMethod>
+  extends RouteConfig<P, Q, B, Method> {
+  validate: Omit<RouteValidatorFullConfig<P, Q, B>, 'body'>;
+  options?: Omit<RouteConfigOptions<Method>, 'body'>;
+}
+
+/**
+ * Kibana Enterprise Search Plugin API endpoints often times pass through the request
+ * body to the Enterprise Search API endpoints for validation. In those cases, we do not
+ * need to validate them in Kibana.
+ *
+ * The safe way to do that is to turn off body parsing entirely using `options.body.parse: false`.
+ * The will pass a String Buffer to the route handler. The proper way to validate this when validation
+ * is enabled to to use `body: schema.buffer()`.
+ *
+ * @see https://github.com/elastic/kibana/blob/master/docs/development/core/server/kibana-plugin-core-server.routeconfigoptionsbody.md
+ * @see https://github.com/elastic/kibana/blob/master/packages/kbn-config-schema/README.md#schemabuffer
+ *
+ * Example:
+ *  router.put({
+ *    path: '/api/app_search/engines/{engineName}/example',
+ *    validate: {
+ *      params: schema.object({
+ *        engineName: schema.string(),
+ *      }),
+ *      body: schema.buffer(),
+ *    },
+ *    options: { body: { parse: false } },
+ *  },
+ *  ...
+ *
+ * This helper applies that pattern, while maintaining existing options:
+ *
+ *  router.put(skipBodyValidation({
+ *    path: '/api/app_search/engines/{engineName}/example',
+ *    validate: {
+ *      params: schema.object({
+ *        engineName: schema.string(),
+ *      }),
+ *    },
+ *  },
+ *  ...
+ */
+export const skipBodyValidation = <Method extends DestructiveRouteMethod>(
+  // DestructiveRouteMethod is the Kibana type for everything except 'get' and 'options'.
+  // Body configuration doesn't apply to those types so we disallow it with this helper.
+  config: ConfigWithoutBodyOptions<unknown, unknown, unknown, Method>
+): Config<Method> => {
```

Comment:
Just have to restate my admiration here. This is awesome and super clean. B E A U T I F U L 😎